integrationtests/tools/proxy/proxy.go

```diff
@@ -200,21 +200,19 @@ func (p *QuicProxy) runConnection(conn *connection) error {
 		}
 		raw := buffer[0:n]
 
-		// TODO: Switch back to using the public header once Chrome properly sets the type byte.
-		// r := bytes.NewReader(raw)
-		// , err := wire.ParsePublicHeader(r, protocol.PerspectiveServer)
-		// if err != nil {
-		// return err
-		// }
+		r := bytes.NewReader(raw)
+		hdr, err := wire.ParsePublicHeader(r, protocol.PerspectiveServer, p.version)
+		if err != nil {
+			return err
+		}
 
-		v := atomic.AddUint64(&conn.outgoingPacketCounter, 1)
+		atomic.AddUint64(&conn.outgoingPacketCounter, 1)
 
-		packetNumber := protocol.PacketNumber(v)
-		if p.dropPacket(DirectionOutgoing, packetNumber) {
+		if p.dropPacket(DirectionOutgoing, hdr.PacketNumber) {
```

Comment:
I'd like to know more about the motivation behind this change. Why is it not sufficient to use the counter for the drop function? Especially with the packet number inferral, I feel like the proxy is becoming quite complicated – and we're going to have to drop all of this anyway when packet number randomization is introduced.

Reply:
Because the interval drop test doesn't work as intended when we don't have the correct packet number. Inferring should be easy here, because there won't be packet loss on localhost.

How do you think packet number randomization will affect this? We can still read the 32 bit packet number on the first packet, and start inferring from there. We will have to do some changes in the drop test, but that's independent.

Reply:
Thinking further about that, we could get rid of the Public Header parsing completely and just pass a "packet count" to the delay and drop callbacks. We won't need any changes to the drop tests then.

Reply:
Yes, that was my intention. I don't see a difference between counts (what we use now), or the actual pnum for the purpose of our tests.